Added test application for Dynamic Rendering

MichaelAnttila opened the pull request:
This test application renders a cube on the screen using the dynamic
rendering extension.  Two new functions (VkCmdBeginRenderingKHR and
VkCmdEndRenderingKHR) were added to the function table in order to
facilitate this.

Tested on Windows with Nvidia.

bjoeris committed:
Fix barrier pipeline stages in async_compute

bjoeris committed:
Format async_compute/main.cpp

bjoeris committed:
async_compute: fix queue transfer barriers

The queue family transfer barriers should use 0 for the stage/access
masks for the stage happening on the other queue family. I.e. the
acquire barriers should use 0 for the srcStageMask and srcAccessMask,
and similarly the release barriers should use 0 for the dstStageMask
dstAccessMask.

bjoeris committed:
Copy async_compute to synchronization2_async_compute

bjoeris committed:
Add synchronization2_async_compute sample

bjoeris committed:
Copy blit_image to synchronization2_blit_image

bjoeris committed:
Add synchronization2_blit_image sample

bjoeris committed:
Copy write_timestamp to synchronization2_write_timestamp

bjoeris committed:
Add synchronization2_write_timestamp sample

bjoeris committed:
Copy set_event to synchronization2_set_event

bjoeris committed:
Add synchronization2_set_event sample

bjoeris committed:
Copy timeline_semaphore_simple to synchronization2_timeline_semaphore_simple

bjoeris committed:
Add synchronization2_timeline_semaphore_simple

elviscapiaq committed:
Added VTA for VK_KHR_format_feature_flags2

RenfengLiu committed:
Add interface to set Vulkan API version (#225)

* Add interface to set Vulkan API version

MichaelAnttila committed:
Added test application for Dynamic Rendering

This test application renders a cube on the screen using the dynamic
rendering extension.  Two new functions (VkCmdBeginRenderingKHR and
VkCmdEndRenderingKHR) were added to the function table in order to
facilitate this.

Tested on Windows with Nvidia.

MichaelAnttila commented:
A future test could expand on this and use the "suspend/resume" features of dynamic rendering.

RenfengLiu commented on application_sandbox/dynamic_rendering_cube/CMakeLists.txt:
```diff
@@ -0,0 +1,31 @@
+# Copyright 2017 Google Inc.
```
nit:2022

RenfengLiu commented on application_sandbox/dynamic_rendering_cube/main.cpp:
```diff
@@ -0,0 +1,318 @@
+            {0}, {}, {VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME}),
+        cube_(data->allocator(), data->logger(), cube_data) {}
+  virtual void InitializeApplicationData(
+      vulkan::VkCommandBuffer* initialization_buffer,
```
nit: it would be nicer if we could call `vkGetPhysicalDeviceFeatures2` with the VkPhysicalDeviceDynamicRenderingFeatures structure included in the pNext chain of the VkPhysicalDeviceFeatures2 to check if the feature is supported or not.

MichaelAnttila replied:
Done.

bjoeris committed:
Fix validation about unsupported pNext structs

These structs are added in Vulkan 1.1, but the samples are still using
Vulkan 1.0 by default, causing validation warnings.

This change only includes the structs in the pNext chain when they are
being activated.

bjoeris committed:
Fix synchronization in async_compute sample

The "mailbox" caused images to be released by the async compute queue
without a corresponding acquire on the render queue, which is illegal.

To fix this issue, whenever the async compute thread sends a buffer to
the mailbox, the existing buffer(s) stay in the mailbox; only the most
recent buffer in the mailbox will actually be rendered, and the
remaining buffers will simply be acquired on the render queue and
immediately released back to the async compute queue.

bjoeris committed:
Add synchronization2 functions to function_table.h

bjoeris committed:
Format write_timestamp/main.cpp

bjoeris committed:
Fix validation errors in write_timestamp

bjoeris committed:
Fix EnableVulkan11 usage in format_feature_flags2

bjoeris committed:
Fix SetvulkanApiVersion

RenfengLiu committed:
Update the version of submodule of google/mathfu. (#240)

The new version now has the https as the its submodules' URL. Previous
mathfu's submodule is using http for submodule and kokoro doesn't
support it.
